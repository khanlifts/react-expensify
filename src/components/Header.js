import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    {/*
      set activeClassName for styling of active link
      set exact={true} to avoid class being appended to home on every page
    */}
    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink><br/>
    <NavLink to="/create" activeClassName="is-active">Create Page</NavLink><br/>
    <NavLink to="/help" activeClassName="is-active">Help Page</NavLink>
  </header>
);

export default Header;
