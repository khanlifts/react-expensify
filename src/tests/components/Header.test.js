import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  // this works because we configured jest.config.json to snapshotSerializers
  // to use enzyme-to-json/serializer
  expect(wrapper).toMatchSnapshot();
});
