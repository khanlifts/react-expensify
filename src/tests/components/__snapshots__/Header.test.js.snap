// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    Create Expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
